python/pyarrow/parquet/__init__.py

```diff
@@ -3453,6 +3471,13 @@ def read_schema(where, memory_map=False, decryption_properties=None):
     n_legs: int64
     animal: string
     """
-    return ParquetFile(
-        where, memory_map=memory_map,
-        decryption_properties=decryption_properties).schema.to_arrow_schema()
+    filesystem, where = _resolve_filesystem_and_path(where, filesystem)
+    source = filesystem.open_input_file(
+        where) if filesystem is not None else where
+    ctx = source if filesystem is not None else nullcontext()
```

Comment:
I find these lines slightly difficult to follow due to using both `source` and `where` variables and the two inline ifs for the same case, would something like:
```
    ctx = nullcontext()
    if filesystem is not None:
        ctx = where = filesystem.open_input_file(where)
```
be slightly more readable?
In thas case we should use `where` instead of `source` on the `ParquetFile` but this was how it was originally too.

Reply:
Makes sense. Thanks for the suggestion. Have updated the code. PTAL :)